### PR TITLE
Have CMake build the dependency graph for linking

### PR DESCRIPTION
**Pull request by kidder**
Also make RunTest independent of stuff in Parallel.
Instead have separate parallel test executables.

### CODE REVIEW

File: tests/Unit/Parallel/Test_Main.cpp [hunk 1]
```diff
@@ -14,19 +14,24 @@
...
 #include "Utilities/TMPL.hpp"
 #include "tests/Unit/Parallel/ParallelTestChares.hpp"
+#include "tests/Unit/Parallel/ParallelTestClasses.hpp"
 #include "tests/Unit/Parallel/ParallelTestTentacles.hpp"
 #include "tests/Unit/TestHelpers.hpp"
 
 #include "tests/Unit/Parallel/Test_Main.decl.h"
 
-void setup_error_handling() {
+ void setup_error_handling() {
```
Comment (nilsdeppe) [inline tests/Unit/Parallel/Test_Main.cpp:23]: Why the space here?

Comment (nilsdeppe) [reply tests/Unit/Parallel/Test_Main.cpp:23]: Don't bother changing the PR, this test is going to change with Algorithm anyway.

**Review by nilsdeppe — APPROVED** [review]

**Comment by nilsdeppe** [comment]
@wthrowe I'm going to wait for you to look at it whenever you get a chance. Please let us know if you have any requested changes or are happy, thanks! :)

**Comment by wthrowe** [comment]
Looks OK to me.  I haven't checked that the listed dependencies are correct, but this looks much more maintainable and apparently still works based on Travis.

**Comment by nilsdeppe** [comment]
Cool, thank you!

**Pull request merged**